### DISCUSSION TITLE
Can not add camera via UI following the documents

### DISCUSSION THREAD

**Discussion by derekkddj** [sca075/mqtt_vacuum_camera#211] [Show and tell]
### Checklist

- [X] I have updated the integration to the latest version available
- [X] I have checked if the problem is already reported

### The problem

Hi, i can not create the camera in via the UI, it give me this error:
In this step of the manual:
![image](https://github.com/user-attachments/assets/e990c2f6-d566-40fe-8594-d2c63c707997)
My home assistant give me this error:

![image](https://github.com/user-attachments/assets/77205779-43c0-4b44-a588-afe170a70035)

### What version of an integration has described problem?

2024.07.3

### What vacuum model do you have problems with?

Roborock.v1

### What type of platform you use?

VEMLinux (Virtual Machine) such as Proxmox, VMWare, etc.

### What version of Home Assistant do you use?

core-2024.7.2

### What type of installation are you running?

Home Assistant OS

### Function, that in your opinion is creating the issue.

Migration Process

**Comment by sca075** [comment]
@derekkddj I did just few minutes ago add a "new vacuum" on 2024.07.3 and all worked fine.
![Screenshot 2024-07-18 at 13 11 04](https://github.com/user-attachments/assets/7b6562f3-acf6-4d02-9c8e-be2b02b40982)
can you please give me more details about it.. I mean the step you used is to add a new mqtt vacuum camera correct? If so I will unstall the camera (remove it from the config) and try to set it up again to see if I can reproduce the issue you are here reporting..
Thanks a lot for it in any cases :)

**Comment by derekkddj** [comment]
Hi, yes, the problem i have is when "adding the camera". Y have the repo installed:
![image](https://github.com/user-attachments/assets/98dfc9b1-97f3-47cf-adf5-941dff790759)

And when i click the link to add the camera:
https://my.home-assistant.io/redirect/config_flow_start/?domain=valetudo_vacuum_camera
It just give me this error:
![image](https://github.com/user-attachments/assets/a4764c32-3ac5-43cc-b820-10f12ad27d97)

I can not see the "camera" in MQTT:

![image](https://github.com/user-attachments/assets/2e6aa7d9-72bd-43fe-9963-bfbb37221eec)

**Comment by sca075** [comment]
@derekkddj

Okay steps followed to reproduce the issue:
1) removed the entities of the camera
2) removed the camera via HACS (so no camera code in custom_componets)
3) Restart Home Assistant to clean the registries.
4) Removed also the valetudo_camera folder from .storage
5) Restarted Home Assistant
6) Check for cameras in home assistant no MQTT Camera Entities listed
-- We start with the setup.
7) Download via HACS the MQTT Vacuum Camera
8) **Restart Home Assistant as it is a Custom Component.**
9) Check MQTT 
![Screenshot 2024-07-18 at 13 32 59](https://github.com/user-attachments/assets/323586d4-50bd-4a00-8716-70140ef1ae39)
10) Add the camera.
![Screenshot 2024-07-18 at 13 35 09](https://github.com/user-attachments/assets/fe1bd8e0-0c09-4a59-9552-229d19818d8d)

**Comment by sca075** [comment]
Domain should be as per already 3 Versions MQTT Vacuum Camera there are over 500 installed cameras so.. question did you had the old Valetudo Vacuum Camera on your system before? if yes this would explain the issue you are facing.
Then please try to use the 2024.06.4 to migrate it and then you can run the 2024.07.3.. 
Hope this helps, if not please just let me know I will be happy to help you.

**Comment by derekkddj** [comment]
OK, adding the cammera in the step 10 worked.
![image](https://github.com/user-attachments/assets/88242b12-960e-46e7-b92f-465470161c07)


But the link that is in the documentation didnt work for me:
https://my.home-assistant.io/redirect/config_flow_start/?domain=valetudo_vacuum_camera

**Reply by sca075** [reply]
> But the link that is in the documentation didnt work for me:
> https://my.home-assistant.io/redirect/config_flow_start/?domain=valetudo_vacuum_camera

@derekkddj  did update the [install procedure](https://github.com/sca075/mqtt_vacuum_camera/blob/main/docs/install.md) and actually checked the docs, from where did you get the above link? I'd like to correct it too but so far I could't find from where it comes actually.

**Reply by derekkddj** [reply]
Hi, i got the link from here:
![image](https://github.com/user-attachments/assets/a4d21629-676a-4d0c-88d9-11c667edac47)

**Reply by sca075** [reply]
Than you so much I will correct it :)

**Comment by sca075** [comment]
I will correct the DOCS soon thanks a lot for confirming it and I'm happy to assist you at any time, do you mind if I move this to the discussions? So that I will inform you as soon the docs are corrected. @derekkddj  thanks once again and if all goes well please do not forget to star the project ;)

**Comment by derekkddj** [comment]
yeah for sure, thank for all the support and fast help :D
